src/Data/Foreign.purs

```diff
@@ -44,16 +44,20 @@ foreign import data Foreign :: *
 
 -- | A type for runtime type errors
 data ForeignError
-  = TypeMismatch String String
+  = TypeMismatch (Array String) String
   | ErrorAtIndex Int ForeignError
   | ErrorAtProperty String ForeignError
   | JSONError String
 
 instance showForeignError :: Show ForeignError where
```

Comment:
There's an informal rule that `Show` should print purescript code, so like `"(TypeMismatch " <> show exp <> show act <> ")"` style, for each case.